kstreams/engine.py

```diff
@@ -132,23 +133,22 @@ async def send(
         return metadata
 
     async def start(self) -> None:
-        await self.start_producer()
-        await self.start_streams()
-
         # add the producer and streams to the Monitor
         self.monitor.add_producer(self._producer)
         self.monitor.add_streams(self._streams)
-        self.monitor.start()
+
+        await self.start_producer()
+        await self.start_streams()
```

Comment:
what was the problem with the order of these ones?

Reply:
There is not a problem, but if we add `anyio` with `TaskGroups` then it think it will be more `decent` to add the `producer` and the `streams` to the monitoring before it starts:

```python
async def _start_streams_on_worker_mode(self, streams) -> None:
    async with anyio.create_task_group() as tg:
        # start monitoring in TaskGroup. Streams and Producers must be already there.
        tg.start_soon(self.monitor.start)
```